src/Controller/ElasticsearchController.php

```diff
@@ -915,33 +806,64 @@ public function searchAction(Request $request)
                 ]);
             }
 
+            $mapIndex = [];
+            if ($response !== null) {
+                $indexes = $response->getAggregation(AggregateOptionService::INDEXES_AGGREGATION);
+                if ($indexes !== null) {
+                    foreach ($indexes->getBuckets() as $bucket) {
+                        $indexName = $bucket->getKey();
+                        if ($indexName === null) {
+                            continue;
+                        }
+                        $aliases = $elasticaService->getAliasesFromIndex($indexName);
+                        foreach ($aliases as $alias) {
+                            if (isset($environments[$alias])) {
+                                $mapIndex[$indexName] = $environments[$alias];
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+
+
             return $this->render('@EMSCore/elasticsearch/search.html.twig', [
-                'results' => $results,
+                'response' => $response ?? null,
                 'lastPage' => $lastPage,
                 'paginationPath' => 'elasticsearch.search',
                 'types' => $types,
-                'alias' => $mapAlias,
                 'indexes' => $mapIndex,
                 'form' => $form->createView(),
                 'page' => $page,
                 'searchId' => $searchId,
                 'currentFilters' => $request->query,
-                'body' => $body,
+                'body' => $searchService->generateSearchBody($search),
                 'openSearchForm' => $openSearchForm,
                 'search' => $search,
                 'sortOptions' => $this->getSortOptionService()->getAll(),
-                'aggregateOptions' => $aggregateOptions,
+                'aggregateOptions' => $aggregateOptionService->getAll(),
             ]);
         } catch (NoNodesAvailableException $e) {
             return $this->redirectToRoute('elasticsearch.status');
         }
     }
 
-    private function getAllContentType($results)
+    /**
+     * @return string[]
+     */
+    private function getAllContentType(CommonResponse $response): array
     {
+        $aggregation = $response->getAggregation(AggregateOptionService::CONTENT_TYPES_AGGREGATION);
+        if ($aggregation === null) {
+            return [];
+        }
         $out = [];
-        foreach ($results['aggregations']['types']['buckets'] as $type) {
-            $out[] = $type['key'];
+        foreach ($aggregation->getBuckets() as $bucket) {
+            $key = $bucket->getKey();
+            if ($key === null) {
+                continue;
+            }
+            $out[] = $key;
```

Comment:
May we should add a function $aggregation->getKeys();

Reply:
Done